flexget/plugins/list/radarr_list.py

```diff
@@ -0,0 +1,378 @@
+    'bluray-576p': 'bluray 576p',
+    'hdtv-720p': 'hdtv 720p',
+    'webdl-720p': 'webdl 720p',
+    'bluray-720p': 'bluray 720p',
+    'hdtv-1080p': 'hdtv 1080p',
+    'webdl-1080p': 'webdl 1080p',
+    'bluray-1080p': 'bluray 1080p',
+    'remux-1080p': 'remux 1080p',
+    'hdtv-2160p': 'hdtv 2160p',
+    'webdl-2160p': 'webdl 2160p',
+    'bluray-2160p': 'bluray 2160p',
+    'remux-2160p': 'remux 2160p',
+    'br-disk': 'remux', # not completely correct
+    'raw-hd': 'remux', # not completely correct
+
+    # No idea of how to map these:
+    # 'regional': 'UNKNOWN'
+}
+
+log = logging.getLogger('radarr_list')
+
+def radarr_quality_to_flexget_quality_req(radarr_quality):
+    """
+    Translates the provided Radarr quality string to a Flexget Requirement instance.
+    Returns None if translation is unsuccessful
+    """
+
+    # QUALITIES_MAP has its keys in lower case
+    radarr_quality = radarr_quality.lower()
+    if not radarr_quality in QUALITIES_MAP:
+        log.warning(
+            'Did not find a suitible translation for Radarr quality \'%s\'',
+            radarr_quality)
+        return None
+
+    flexget_quality_req_string = QUALITIES_MAP[radarr_quality]
+
+    try:
+        return Requirements(flexget_quality_req_string)
+    except ValueError:
+        log.error(
+            'Failed to convert %s into a valid quality requirement',
+            flexget_quality_req_string)
+        return None
+def get_flexget_qualities(profile, cutoff_only=False):
+    quality_requirements = []
+    if cutoff_only:
+        name = profile['cutoff']['name']
+        quality_req = radarr_quality_to_flexget_quality_req(name)
+        if not quality_req is None:
+            quality_requirements.append(quality_req)
+    else:
+        for quality in profile['items']:
+            if quality['allowed']:
+                name = quality['quality']['name']
+                quality_req = radarr_quality_to_flexget_quality_req(name)
+                if not quality_req is None:
+                    quality_requirements.append(quality_req)
+
+    return quality_requirements
+
+
+
+class RadarrSet(MutableSet):
+    """ Accesses the Radarr movies using the provided the config """
+
+    def __init__(self, config):
+        self.config = config
+        self.service = RadarrAPIService(
+            config['api_key'],
+            config['base_url'],
+            config['port']
+            )
+
+        # Class member used for caching the items to avoid
+        # unnecessary calls to the Radarr API.
+        # We use the self.items property to access it.
+        # Just set this to None again later to invalidate the cache.
+        self._movie_entries = None
+
+    def __iter__(self):
+        return iter(self.items)
+
+    def __len__(self):
+        return len(self.items)
+
+    def discard(self, value):
+        if not value:
+            return
+
+        if not isinstance(value, Entry):
+            raise TypeError()
```

Comment:
Well, I thought, because nothing in Python is really typed I would need to make sure that the item provided is of the correct type in runtime to avoid unhandled exceptions later. But I can't see any other plugin doing these checks so I will just remove them in terms of consistency with the rest of the code base.